---
title: "Group Project: Investigating the validity of *Mnemiopsis leidyi* as a model organism for the study of multicellularity"
author: "Charlotte Barclay and Gabriel Dall'Alba"
date: "31/03/2021"
output:
  github_document:
    toc: true
    toc_depth: 4
---
## 1 Introduction

### 1.1 Motivation

From ‘primordial soup’ to the vast array of biodiversity we see today, evolution and these origins of multicellularity still fascinate and elude scientists[reference 1].  Accounts of the number of independent events that led to multicellularity differ amongst the scientific community[reference 2], although there is a consensus that this happened once in the Animalia/Metazoan lineage[references 2,3]. Advances in sequencing and reduction in cost has led to increase in whole genomic sequences of non-bilateran animal species providing insight into the molecular mechanisms that govern multicellularity[reference 4].

Ctenophores have been proposed as a model organism for understanding the evolutionary mechanisms of multicellularity in animals[references 4,5], however their phylogenetic placement is still widely debated[references 5–7]. Ctenophores are a gelatinous phylum of marine metazoans with approximately 150 known species that form a clade of pre-bilateran animals[reference 8].

![Schema of phylogenetic position of Mnemiopsis leidyi](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/Mle.jpg)

### 1.2 Original Study

overview of the original study, the scope of your re-analysis, and why you chose it

##### 1.2.1 Sample collection and Genome Assembly

Ryan et al collected 2 wild animals from the Vineyard Sound near Woods Hole, Massachussets (Figure 2) [Ryan paper ref]. Those animals were self-fertilized and DNA was isolated from the resulting embryos of one of them. Details of the DNA isolation protocol were not disclosed, but the authors mention the use of "GS FLX Titanium Rapid Library Preparation Kit" and the "GS FLX Titanium Library Paired End Adaptors Kit". The resulting isolated DNA was used for sequencing using a Roche 454 Genome Sequencer FLX machine located at the Roche Applied Science centre in Indianapolis, IN. 7,334,972 raw reads with an Average read length of 339 bases were generated in nine runs, yielding 2.5 Gb of sequence.

![Location where wild ctenophores were collected.](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/woodshole.png)

Those raw reads were then submitted for assembly using the Phusion assembler [phusion reference], resulting in 24,884 contigs with a total of 150,340,428 bases and a reported N50 of 11,936 bases. The authors proceeded to sequence the embryos of the second wild animal using Illumina GA-iiX system. The resulting paired-end reads were filtered, mapped to the 24,884 assembled contigs using Illumina's short read aligner ELAND, and integrated into Phusion's scaffolding step. This resulted in 5,100 scaffolds with an N50 of 187 Kb and an coverage of 160X.

#### 1.2.2 Evaluation of completeness and correctness of genome assembly

The authors extracted 15,752 M. leidyi Expressed Sequence Tags (ESTs) and aligned them to their assembled genome using BLAT [Blat ref] v. 34x12 with default parameters. They evaluated their alignment using a software developed by their group called baa.pl [baa.pl ref] later reformed into Isoblat [github link to Isoblat] and obtained:

1) 99.4% of the transcripts were mapped with BLAT 
2) 98.2% of the positions in the mapped transcripts were aligned
3) 95.2% of the transcripts mapped to a single scaffold.  

They further extracted and sequenced 79 Mb paired-end/83 Mb single-end RNA reads from M. leidyi embryos through Illumina GA-II platform and assembled them using Trinity [Trinity Ref]. Their final assembly had 32,630 for paired-end and 27,315 for single-end transcripts respectively. They proceeded to align their Trinity-assembled transcripts to BLAT and analyze it through baa.pl with default parameters and generated the following statistics:

1) 99.2% of the transcripts were mapped with BLAT
2) 98.1% of the positions in the mapped transcripts were aligned
3) 92.7% of the transcripts mapped to a single scaffold.

Thus, they conclude that their assembly is likely complete and correctly assembled. Further repeat analysis indicates allowed them to estimate a genome size of 150 Mb, which configures this genome to be one of the smallest 7% known genomes [Ryan ref].

#### 1.2.3 Gene Prediction pipeline

The authors provided a complex, in-depth pipeline for gene prediction with no common pattern between automated and manual steps. Firstly, the authors sequenced another round of RNA-seq data, resulting in 162 Mb of RNA-seq data originating from mixed stage embryos (ranging from a few hours up until 15 hours post-fertilization). These reads were mapped to their assembled genome using TopHat [tophat ref] and then assembled into 49,850 transcript fragments using Cufflinks [cufflink ref].

They then loaded  the 49,850 Cufflink fragments, the 15,752 publicly available ESTs, and additional 161 alleged publicly available cDNA sequences into PASA [pasa ref] without providing details of this step. They masked the genome with a repeat library, excluding from the masking regions where RNA-seq mappings overlapped, resulting in 53,244 regions and 7,387,140 base pairs of sequence. This data was then submitted to several prediction softwares: GENESH [genesh ref]; AUGUSTUS [augustus ref] (version_2.3.1); HMMgene [hmmgene ref] (version_1.1); and GenomeScan [genomescan ref] (version_0.1). They evaluated the predicted models using Evidence Modeler (EVM version_r03062010) [EVM ref]. In total, the authors established that the genome contains 16,548 predicted genes (or protein-coding loci), making up 58% of its total length, with 44% of those loci being homologous to known genes in non-ctenophores.   

#### 1.2.4 Phylogenetic analysis

### 1.3 Re-analysis

This study was chosen, as the origins of multicellularity and particularly the phylogenetic position of Ctenophores compared to other non-bilaterans is still widely debated. The original study was undertaken in 2011 and published in 2013 [ref 5?], since then of the tools the paper used, some have changed, some are no longer maintained and more data has been made available for gene annotation.

In our re-analysis, in the absence of raw reads we will investigate the files to confirm number of contigs and scaffolds before running the following analysis to assess for completeness of the assembly:
 - Replicate and validate the alignment of ESTs as seen in the original study using BLAT
 - Replicate and validate the alignment of transcripts as seen in the original study using BLAT
 - Compare alignment of ESTs with original tool BLAT and Bowtie
 - Design Perl script to assess GC content in absence of fastqc files
 
Following this assessment for correctness of assembly, we will undertake the following analysis:
 - Annotation
 - Phylogeny


## 2 Workflow: Methods and Results

integrated data processing, QC, analysis, results, graphs, and other data, as well as written explanations for what is being done and why, and interpretations of results. This should flow in chronological order (e.g. starting with fastqs and ending with the last graph).

### 2.1 Data gathering

#### 2.1.1 Mnemiopsis Genome Portal

The data was accessed from the Mnemiopsis Genome Portal [Genome Portal ref], a public database for genomic and functional information on *M. leidyi*. The Portal is regularly maintained, with a BLAST interface as well as a visualization tool which allows exploration of the scaffolds and RNA seq data, similarly to the IGV Browser [ref 4?]. Alternatively the data from this study can be obtained from GenBank (here)[https://www.ncbi.nlm.nih.gov/assembly/GCA_000226015.1]. The files from both resources are the same, scaffold and contig .fasta files. Unfortunately after much searching for  raw reads, that data is not publicly available and in order to get the raw data we would need to get this directly from the authors which was not feasible in the given time. Instead the genome was downloaded from: https://research.nhgri.nih.gov/mnemiopsis/download/download.cgi?dl=genome.

**LOADING GENOME**

```{bash, eval=FALSE}

#Download the scaffolds to local machine then imported to server using pscp

pscp -P 22 -r C:\Users\heavy\Documents\bmeg-591\BMEG-591-Rep\Project\MlScaffold09.nt.gz gdalba@gi-edu-sv2.bme.ubc.ca:/home/gdalba/MlScaffold09.nt.gz

#OR Download the contigs directly to server using wget - This comes from the alternative source, not the Genome Portal.

wget -o ./Project/AGCP01.zip https://sra-download.ncbi.nlm.nih.gov/traces/wgs03/wgs_aux/AG/CP/AGCP01/AGCP01.1.fsa_nt.gz

#On server, unziped using gzip

gzip -d MlScaffold09.nt.gz

#We confirmed the 5,100 scaffolds through:

grep -c "^>" MlScaffold09.nt
5100

```
 
**LOADING ESTs**

We also downloaded the 15,752 publicly available ESTs from:  https://research.nhgri.nih.gov/mnemiopsis/download/download.cgi?dl=est.

```{bash, eval=FALSE}

#Imported to server using pscp

pscp -P 22 -r C:\Users\heavy\Documents\bmeg-591\BMEG-591-Rep\Project\MlESTs.gz gdalba@gi-edu-sv2.bme.ubc.ca:/home/gdalba/MlESTs.gz

#On server, unziped using gzip

gzip -d MlESTs.gz

#We confirmed the 15,752 ESTs through:
grep -c "^>" MlESTs.fa
15752

```

**LOADING RNA DATA**

35,203 assembled transcripts using Trinity were downloaded from https://research.nhgri.nih.gov/mnemiopsis/download/download.cgi?dl=transcript and loaded into the server.

```{bash, eval=FALSE}

#Imported to server using pscp

pscp -P 22 -r C:\Users\heavy\Documents\bmeg-591\BMEG-591-Rep\Project\Ml_Trinity_transcripts.fa.gz gdalba@gi-edu-sv2.bme.ubc.ca:/home/gdalba/Ml_Trinity_transcripts.fa.gz

#On server, unziped using gzip

gzip -d Ml_Trinity_transcripts.fa.gz

#We confirmed the 15,752 ESTs through:
grep -c "^>" Ml_Trinity_transcripts.fa
35203

```


### 2.2 Evaluation of Correctness
#### 2.2.1 Bowtie x BLAT

**BLAT**

**BLAT INSTALLATION**

BLAT v. 36x2 was installed through our conda environment:

```{bash, eval=FALSE}

conda activate gdalba-conda

conda install ucsc-blat

```

**RUNNING BLAT**
We ran BLAT with default parameters as described in the paper as follows:

```{bash, eval=FALSE}

blat MlScaffold09.nt MlESTs mle_blatalign.psl

Loaded 155865547 letters in 5100 sequences
Searched 10552828 bases in 15752 sequences
mle_blatalign.psl

```

**BAA.PL/ISOBLAT INSTALLATION**

We cloned baa.pl/Isoblat github repo (https://github.com/josephryan/isoblat) on our own machines and imported it to our server:

```{bash, eval=FALSE}

#Importing baa.pl/Isoblat to server

pscp -P 22 -r C:\Users\heavy\Documents\bmeg-591\BMEG-591-Rep\Project\baa.pl\isoblat gdalba@gi-edu-sv2.bme.ubc.ca:/home/gdalba/baa.pl

#Installing baa.pl/Isoblat

perl Makefile.PL
make
make install

```

**RUNNING BAA.PL/ISOBLAT**

We ran baa.pl/Isoblat with default parameters as described in the paper as follows:
```{bash, eval=FALSE}

isoblat --max_gap_to_consider_missing=5 --min_to_count_as_coverage=5 mle_blatalign.psl MlESTs.fa 2>/dev/null

# running version 0.31 of isoblat
# run with this command: /home/gdalba/.conda/envs/gdalba-conda/bin/isoblat mle_blatalign.psl MlESTs.fa
# $max_gap_to_consider_missing = 5
# $min_to_count_as_coverage    = 5

Ratio of transcripts with a BLAT entry (15665/15752): 0.99447689182326

```

As it can be seen, only one of the three expected statistics were yielded as output. We are unsure of the reasons why baa.pl/Isoblat algorithm fails to capture enough information for the other two calculations (% of transcripts positions successfully aligned and % of transcripts that mapped to a single scaffold). We attempted to troubleshoot the algorithm itself to no luck. The algorithm attempted to read non-existent lines, returning non-numeric values and divisions either by non-numeric values or by zero.

**RUNNING BLAT + BAA.PL/ISOBLAT FOR RNA DATA**
We repeated the above steps using Trinity-assembled transcripts:

```{bash, eval=FALSE}

blat MlScaffold09.fa Ml_Trinity_transcripts.fa blatrnaalign.psl
Loaded 155865547 letters in 5100 sequences
Searched 29298205 bases in 35203 sequences

isoblat /home/gdalba/blatrnaalign.psl /home/gdalba/Ml_Trinity_transcripts.fa
# running version 0.31 of isoblat
# run with this command: /home/gdalba/.conda/envs/gdalba-conda/bin/isoblat /home/gdalba/blatrnaalign.psl /home/gdalba/Ml_Trinity_transcripts.fa
# $max_gap_to_consider_missing = 5
# $min_to_count_as_coverage    = 5
Ratio of transcripts with a BLAT entry (34815/35203): 0.988978212084197

```

**BOWTIE2**

We also attempted the same correctness evaluation using a different aligner - Bowtie2 [bowtie2 ref]. We performed steps similarly as to what was described in class. The main goal was to comparte different alignment tools and their output. Bowtie2 was already installed server-side.

**CREATING AN INDEX**

```{bash, eval=FALSE}
bowtie2-build MlScaffold09.nt /home/gdalba/genome_index/Mlindex
```

**RUNNING BOWTIE2 WITH ESTs**

```{bash, eval=FALSE}
bowtie2 -x /home/gdalba/genome_index/Mlindex -f MlESTs -S /home/gdalba/mle_align.index

15752 reads; of these:
  15752 (100.00%) were unpaired; of these:
    6989 (44.37%) aligned 0 times
    8562 (54.36%) aligned exactly 1 time
    201 (1.28%) aligned >1 times
55.63% overall alignment rate

```

**RUNNING BOWTIE2 WITH TRINITY-ASSEMBLED RNA TRANSCRIPTS**

```{bash, eval=FALSE}
bowtie2 -x /home/gdalba/genome_index/Mlindex -f Ml_Trinity_transcripts.fa -S /home/gdalba/mle_rnaalign.index


35203 reads; of these:
  35203 (100.00%) were unpaired; of these:
    16346 (46.43%) aligned 0 times
    17177 (48.79%) aligned exactly 1 time
    1680 (4.77%) aligned >1 times
53.57% overall alignment rate

```

### 2.3 ASSESSING GC CONTENT

We employed 2 distinct GC content  measurement approaches: a manual approach and a pre-built script.

**BIOBASH SCRIPTS**

We cloned the Github repo (https://github.com/gibbslab/biobash) containing several general utilities Bioinformatics scripts called BIOBASH, developed in Bash language. We imported the scripts using pscp and converted scripts of interest using the command "dos2unix -k -o <filename>":

**BIOBASH INSTALLATION**

```{bash, eval=FALSE}

-P 22 -r C:\Users\heavy\Documents\bmeg-591\BMEG-591-Rep\Project\biobash gdalba@gi-edu-sv2.bme.ubc.ca:/home/gdalba/scripts

dos2unix -k -o installbiobash
chmod +x installbiobash
./installbiobash

```

To test if the BIOBASH set os scripts were working, we repeated the validation of assembly file MlScaffold09.nt (converted into fasta) with bb_count_seqs script (script that count sequences) after troubleshooting and fixing minimal problems within the script:

```{bash, eval=FALSE}

dos2unix -k -o bb_count_seqs
chmod +x bb_count_seqs
./bb_count_seqs /home/gdalba/MlScaffold09.fa
5100 Scaffolds

```

We then ran bb_get_GC_content after troubleshooting and fixing minimal problems within the script, which required a .fa file with a single header:

```{bash, eval=FALSE}

grep -v ">" MlScaffold09.fa> > MlScaffold09_singleseq.fa

```

```{bash, eval=FALSE}

dos2unix -k -o bb_get_GC_content
chmod +x bb_get_GC_content
./bb_get_GC_content /home/gdalba/MlScaffold09_singleseq.fa
37

```

bb_get_GC_content reported a GC content of 37%. We then applied two perl approaches 

**PERL: GC CONTENT**

```{bash, eval=FALSE}

perl -lane 'unless (/^>/) { $l += length(); $gc++ while /[GC]/ig } END { print $gc/$l}' /home/gdalba/MlScaffold09.fa
0.374857844626818

```

**PERL: GC CONTENT PER SCAFFOLD**

```{bash, eval=FALSE}

perl -lane 'if (/^>(.+)/) {
 printf "%s %d %.2f\n", $n, $l, $gc/$l if $l;
 $n = $1; $l = 0; $gc = 0;
} else {
 $l += length($_);
 $gc++ while /[GC]/ig;
} END {
 printf "%s %d %.2f\n", $n, $l, $gc/$l if $l;
}' /home/gdalba/MlScaffold09.fa

Resulting Average: 0,3754960784
```

**Plot GC content**

```{R, eval=TRUE}
library(readxl)
library(ggplot2)

#Import the data
Gc_per_scaffold <- read_excel("C:/Users/cbarc/Documents/CB_BMEG591E-repository/Group_Project/GC_Per_Scaffold.xlsx",
                            col_types = c("text", "numeric", "numeric",
                                          "skip", "skip", "skip", "skip", "skip"))



#Histogram with the average plotted
ggplot(Gc_per_scaffold, aes(Values)) +
  geom_histogram(binwidth= 0.01, colour = "black", fill="grey")+
  geom_vline(aes(xintercept=mean(Values)),
             color="blue", linetype="dashed", size=1)

#Density and average line. NOTE y axis is density rather than Count
ggplot(Gc_per_scaffold, aes(Values)) +
  geom_histogram(aes(y=..density..), colour="black", fill="white",  binwidth= 0.01)+
  geom_density(alpha=.2, fill="#FF6666") +
  geom_vline(aes(xintercept=mean(Values)),
              color="blue", linetype="dashed", size=1)
```


### 2.4 Annotation

We installed Augustus and its dependencies through conda:

```{bash, eval=FALSE}

conda install -c bioconda augustus
conda install -c anaconda libboost

```

We prepared a hints file using the ESTs aligned to the genome (through the BLAT output) with aid of the following tutorial: https://vcru.wisc.edu/simonlab/bioinformatics/programs/augustus/docs/tutorial2015/prediction.html. Originally, the authors loaded the *M. leidyi* RNA-seq reads (49,850 Cufflinks transcripts), publicly available mRNAs (161) and publicly available ESTs (15,752). We loaded the Cufflink Transcripts and the publicly available ESTs through the following:

```{bash, eval=FALSE}

#Ran filterPSL to exclude alignments below 80% of query length coverage

#mle_blatalign.psl = BLAT alignment of ESTs to the assembled genome.
#blatrnaalgin.psl = Trinity assembled transcripts
#mle_cuffalign.psl = Cufflinks assembled transcripts

cat mle_blatalign.psl | perl /home/gdalba/.conda/pkgs/augustus-3.2.2-0/scripts/filterPSL.pl --best --minCover=80 > filterMle_blatalign.psl

cat blatrnaalign.psl | perl /home/gdalba/.conda/pkgs/augustus-3.2.2-0/scripts/filterPSL.pl --best --minCover=80 > filterMle_rnaalign.psl

cat mle_cuffalign.psl | perl /home/gdalba/.conda/pkgs/augustus-3.2.2-0/scripts/filterPSL.pl --best --minCover=80 > filterMle_cuffalign.psl

#As we can see, this filtering step reduced the amount of relevant alignments from 165481 to 14505, 318457 to 31271, and 2440089 to 50107 for mle_blatalign.psl, blatrnaalign.psl, and mle_cuffalign.psl respectively.

wc -l mle_blatalign.psl filterMle_blatalign.psl
  165481 mle_blatalign.psl
   14505 filterMle_blatalign.psl
  179986 total
  
wc -l blatrnaalign.psl filterMle_rnaalign.psl
  318457 blatrnaalign.psl
   31271 filterMle_rnaalign.psl
  349728 total

wc -l mle_cuffalign.psl filterMle_cuffalign.psl
  2440089 mle_cuffalign.psl
    50107 filterMle_cuffalign.psl
  2490196 total

#We then merge the two filtered files.

cat filterMle_blatalign.psl filterMle_cuffalign.psl > combined_cuffhints.psl

#And run blat2hints

blat2hints.pl --nomult --in=combined_cuffhints.psl --out=hints.mle.gff

```

With the hints file ready, we can run Augustus. Augustus was originally trained using *Amphimedon queenslandica*, a data set that is already available through the tool itself.


```{bash, eval=FALSE}

augustus --progress=true --species=amphimedon --extrinsicCfgFile=extrinsic.ME.cfg --hintsfile=hints.mle.gff MlScaffold09.fa > augustus_annotation.gff

where,
--progress=true shows the performance of every step

--species=amphimedon indicates the training dataset, alternatively, you would train your own dataset.

--extrinsicCfgFile=extrinsic.ME.cfg loads a .cfg file containing metadatafrom the hints file, for instance, and parameters that help to assess the quality of the gene prediction and accurately decide where which gene elements are in a predicted sequence. The .ME.cfg file takes into account information coming from the ESTs located within the Hints file, which is useful in our case.

--hintsfile=hints.mle.gff specifies the hints we generated in the previous steps.

By default, Augustus predicts on both forward and reverse.
```

```{bash, eval=FALSE}

#Assessing number of predicted genes by Augustus

awk '$3=="gene"' augustus_annotation.gff | wc -l
33354

Example of Augustus output:

ML5014  AUGUSTUS        gene    20      829     0.87    +       .       g11104
ML5014  AUGUSTUS        transcript      20      829     0.87    +       .       g11104.t1
ML5014  AUGUSTUS        start_codon     20      22      .       +       0       transcript_id "g11104.t1"; gene_id "g11104";
ML5014  AUGUSTUS        CDS     20      829     0.87    +       0       transcript_id "g11104.t1"; gene_id "g11104";
ML5014  AUGUSTUS        stop_codon      827     829     .       +       0       transcript_id "g11104.t1"; gene_id "g11104";
# protein sequence = [MLQDSRSYAGTLAYSDHRLVVTRVNFKNICLCYKRHTQSSFKFDTSELTSNPTIQVKYRHSLNENLSNAVPALDPNSD
# LNSLLESIKDAAKSSIGVTRHRKRNRSDNDEVKSLSNQRHMLRQQLNNNQSMDRTLLRSSINRLTNQIQQRLSILRSAAADAICNTISTTTDSKKMFE

We can run this prediction against Mnemiopsis protein models using BLASTp (https://research.nhgri.nih.gov/mnemiopsis/sequenceserver/#), the best match (for this protein, is the protein model ML050010a, with an e-score of 6e-122) can then undergo regular Blastp analysis, revealing 83% identity (e-value: 4e-43) with an uncharacterized protein found in *Octopus sinensis*.

```

Multiple attempts to run PASA, HMMGene, and GENESH were made, but due to time constrains and technical challenges, we decided to run Augustus alone. Augustus predicted 33,354 protein-coding loci. We then used BED-tools getfasta function to extract sequences:

```{bash, eval=FALSE}
#filters every possible sequence type into a .bed file

awk '($3=="CDS" || $3=="gene" || $3=="tRNA" || $3=="tmRNA" || $3=="ncRNA" || $3=="rRNA") {OFS="\t"; print $1,$4-1,$5}' augustus_annotation.gff > mlegenes.bed

#getfasta function correlates predicted sequences in .bed file back to the assembled genome and store them on a .ffn file

bedtools getfasta -fi MlScaffold09.fa -bed mlegenes.bed -fo mle.ffn


```

When mapping back to the genome, we obtain 29,875 genes. This file can then be used to manually curate sequences following the previously mentioned BLASTp approach.


### 2.5 Differential Expression

First we download the temporal development expression profiles to from (GenBank)[https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60478]. The introns of the genes are removed to form the mature mRNAs, so the reads that are counted correspond to the reads aligning to the exons of the genes.

Load relevant libraries as required

```{r, eval=FALSE}
# Load library for DESeq2
library(DESeq2)

# Load library for RColorBrewer
library(RColorBrewer)

# Load library for pheatmap
library(pheatmap)

# Load library for tidyverse
library(tidyverse)
```

```{r, eval=FALSE}
#load the data
load("GSE60478_RNAseq.RData") 

#explore the counts data
head(GSE60478_ML_exp.tab)

#explore the metadata
head(GSE60478_ML_GEO_sample_sheet.tab) 
```

```{r, eval=TRUE}

```


### 2.5 Phylogeny

In the original study, phylogenetic analysis was conducted using PhyloBayes "a Bayesian Monte Carlo Markov Chain (MCMC) sampler for phylogenetic
reconstruction", comparing the whole genomes of 13 different animals in the 'Genome set' vs the EST data available in GenBank from 58 different animals in the 'EST set'. Unfortunately, one of the runs for EST alone took 204 days and as such was not reproducible for this assignment.

Phylogenetic analysis allows researchers the opportunity to identify potential similarity between species, indicating lineage [REF]. In conjunction with annotation, phylogeny could be used to assess orthologs and genetically conserved regions that can be traced back to the root of metazoan divergence. Consequently, instead of replicating the whole genome or EST phylogenetics (due to the aforementioned time limitations), we compared significant gene regions that are of interest to scientists in the origins of animal multicellularity. 

In the original study, Ryan et al. suggested through phylogenetic analysis that ionotropic glutamate receptors from *M. leidyi* form a sister clade to the bilaterian glutamate receptors. The gene model of interest was identified as ML00441a and both the transcript sequence and predicted protein sequence can be viewed [here](https://research.nhgri.nih.gov/mnemiopsis/wiki/index.php/ML00441a).


#### 2.5.1 Mnemiopsis Genome Project Portal: Track Viewer

The following is a snap shot from the Mnemiopsis leidyi genome portal track for the gene region of interest. This viewer can be interrogated similarly to the IGV. 

![Mnemiopsis leidyi Genome Portal Track for ML00441a](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/Track.png)

This region of interest can be further zoomed in, in order to see the more information from the Reference sequence as well:

![Mnemiopsis leidyi Genome Portal Track for ML00441a](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/Track_Zoom.png)

#### 2.5.2 BLAST 

We extracted the nucleotide sequence, as well as the protein sequence for the target gene region ML00441, as identified in the supplementary material as a ionotropic glutamate receptor region. Using [NCBI BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi), we blasted the first the nucleotide, followed by the protein sequence to identify similar sequences. The top results were ionotropic glutamate receptors for other ctenophore species. 

BLAST describe the E value (the expect value) as the random chance we could "expect" to get a hit when searching the database of a particular size, in other words the E value describes the random background noise. The lower the E-value (the closer it is to zero), the more "significant" the match is[REF](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=FAQ#:~:text=The%20Expect%20value%20(E)%20is,describes%20the%20random%20background%20noise). The results can be seen below:

**Nucleotide Blast search results** </br>

![Nucleotide Blast Search results](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/BLASTN_ML00441a.png)

**Nucleotide Blast search results** </br>

![Protein Blast Search results](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/BLASTP_ML00441a.png)

#### 2.5.3 Create a Phylogenetic Tree using FigTree.

Firstly we downloaded the .fasta protein sequences identified above into a single file. Then we retrieved published sequences from GenBank for a number of AMPA (GRIA) and delta2-like (GRID) glutamate receptors, as described in the original study. These protein sequences (GRID1 (Accession AAH39263.1), GRID2 (Accession AAH99654.1), GRIA1 (Accession XP_016864881.1), GRIA2(Accession AAH10574.1)) were also added to .fasta file along with the sequence for ML00441a.

Secondly, we generated a multiple sequence alignment using the EBI tool *MU*ltiple *S*equence *C*omparison by *L*og - *E*xpectation , known as [MUSCLE](https://www.ebi.ac.uk/Tools/msa/muscle/), with the output defined as Pearson/FASTA.

```{bash, eval=FALSE}
conda install -c bioconda muscle
muscle -in ./BLAST_P.fasta -quiet -fasta -out BLASTP_Aligned.fasta
```

Finally we ran a Neighbor Joining analysis using the EBI phylogeny tool [Simple Phylogeny](https://www.ebi.ac.uk/Tools/phylogeny/simple_phylogeny/), which was used as the input to the FigTree software to generate the tree. FigTree was used to colour annotate the tree. FigTree is available [here](http://tree.bio.ed.ac.uk/software/figtree/).

```{bash, eval=FALSE}
conda install -c bioconda muscle
muscle -in ./BLAST_P.fasta -quiet -fasta -out BLASTP_Aligned.fasta
/nfs/public/ro/es/appbin/linux-x86_64/clustalw-2.1/bin/clustalw2 -infile=simple_phylogeny-I20210422-022731-0145-87347698-p1m.upfile -tree -outputtree=phylip -clustering=Neighbour-joining
```

![Phylogenetic Tree](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/BLAST_PTree.jpg)

## 3. Discussion

Here, we attempted to familiarize with most of the fundamental steps of the *M. leidyi* sequencing project. The gap between sample collections and publishing of the Genome Portal/Science paper indicates that it took a minimum of two years to successfully assemble and annotate the genome of this relatively unknown early metazoan. Our attempts to replicate the BLAT alignment were partially successful, where we reported that 99.4% and 98.8% of the ESTs and transcripts respectively were mapped with BLAT. Indeed, the authors provided the scores of 99.4% and 99.2% (we consider this margin to be negligible) for the same statistic. Due to issues while running baa.pl/Isoblat, the other two meaningful statistics were not generated on our end. We speculate that this is due to incompatibilities of the tool with its dependencies - Isoblat was last updated three years ago. Nowadays, tools like GenomeQC [GenomeQC ref] are often used for assessment of assembly quality [maybe add a review ref that says the same?]

We attempted to replicate the alignment using Bowtie2. Surprisingly, Bowtie2 alignment scores fell under 60% (55.63% and 53.37% for ESTs and transcripts alignment to the genome respectively). We attempted to assess the reasons that might explain the differences. BLAT is a tool that operates pretty much like BLAST [BLAST ref], but there are a few differences in its structure. While BLAST targets GenBank sequences, BLAT indexes the target genome, similarly to Bowtie2. BLAT indexing approach is known as hash-based - a data structure format that associate strings to a hash value through hash functions, resulting in a hash table [hash ref], and the tool builds this index by finding all the non-overlapping 11-mers not heavily involved in repeats [BLAT ref]. Bowtie2, on the other hand, uses an FM-index based on a Burrows-Wheeler Approach [bowtie2 ref] through an algorithm known as Blockwise [juha ref]. 

In spite of indexing being a compression step to speed up alignment, it is known that different compression algorithms and different data structure approaches can yield slightly differences in output. However, this is unlikely a reasonable explanation for the big discrepancies between Bowtie2 and BLAT outputs. However, BLAT applies a gap-aware approach, whereas Bowtie2 does not. Blat can also recognize and work around introns and exons, being generally designed to work better with the alignment of ESTs to a assembled genome. That could be a reasonable explanation, and it highlights that alignment softwares should be chosen according to the researcher's main goals. For this specific dataset, it is unlikely that Bowtie2 is able to yield meaningful results, whereas BLAT clearly is able to.

We report genomic GC content at around 37% in all our measurement approaches. The authors report 38.86%. We assume the 1% discrepancy comes from the extra steps the authors employed to mask repeats detected using RepeatMasker [repeatmasker ref], and conclude that we achieve a satisfactory match, given this condition.

Our genome annotation attempt using Augustus missed a few data processing steps they authors initially made, including masking repeats and incorporating additional 161 cDNA sequences. The authors reported 29,359 predicted protein-coding loci. In comparison to HMMGene (that predicted 13,948 genes), it predicted 15,411 additional loci. Augustus consistently predicted a higher number of loci than every other employed tool. By the end, the authors opted for keeping FGENESH and PASA predictions, evaluated through EVM (EvidenceModeler) [EvidenceModeler ref], obtaining 16,845 genes, closer to the reported final number of 16,548 genes after manual curation of additional sequences.

Nevertheless, Augustus initial value predicted 33,354 genes, predicting aprox. 4,000 genes more than the expected. Intriguingly, when mapping the sequences back to the assembled genome using BED-tools, we obtain 29,875 genes, a value much closer to their reported value  As described, this increment is likely caused by the missing steps (and for the final value, likely cause by the missing 161 CDNA sequences incorporated on the hints file). Unsurprisingly, running our Augustus prediction sequences against the reference genome via BLASTp reveals that the predicted genes match consistently known annotated *M. leidyi* genes (we were able to consistently find sequences scoring high identity and significant e-value).

## Conclusion

Here, we attempted to partially cover a 2-year long *M. leidyi* genome sequencing project. We were able to 


## Bibliography

1.	Van Gestel J, Tarnita CE. On the origin of biological construction, with a focus on multicellularity. Proc Natl Acad Sci U S A. 2017;114(42):11018-11026. doi:10.1073/pnas.1704631114
2. 	Niklas KJ, Newman SA. The many roads to and from multicellularity. J Exp Bot. 2020;71(11):3247-3253. doi:10.1093/jxb/erz547
3. 	Niklas KJ, Newman SA. The origins of multicellular organisms. Published online 2013. doi:10.1111/ede.12013
4.  Moreland RT, Nguyen AD, Ryan JF, Baxevanis AD. The Mnemiopsis Genome Project Portal: Integrating new gene expression resources and improving data visualization. Database. 2020;2020(1):1-9. doi:10.1093/database/baaa029
5. 	Ryan JF, Pang K, Schnitzler CE, et al. The genome of the ctenophore Mnemiopsis leidyi and its implications for cell type evolution. Science (80- ). 2013;342(6164). doi:10.1126/science.1242592
6. 	Ryan JF, Pang K, Mullikin JC, Martindale MQ, Baxevanis AD. The homeodomain complement of the ctenophore Mnemiopsis leidyi suggests that Ctenophora and Porifera diverged prior to the ParaHoxozoa. Evodevo. 2010;1(1):1-18. doi:10.1186/2041-9139-1-9
7. 	Ryan JF. Did the ctenophore nervous system evolve independently? Zoology. 2014;117(4):225-226. doi:10.1016/j.zool.2014.06.001
8. 	Moroz LL, Kocot KM, Citarella MR, et al. The ctenophore genome and the evolutionary origins of neural systems. Nature. 2014;510(7503):109-114. doi:10.1038/nature13400

