---
title: "Group Project: Investigating the validity of *Mnemiopsis leidyi* as a model organism for the study of multicellularity"
author: "Charlotte Barclay and Gabriel d'Alba"
date: "31/03/2021"
output:
  github_document:
    toc: true
    toc_depth: 4
---

**Install BLAT and abyss**

Abyss is an assembly tool as could not find Phusion package.

```{bash, eval=FALSE}
conda install -c bioconda ucsc-blat
conda install -c bioconda abyss
```


```{r, eval=FALSE}
pscp -P 22 C:\Users\cbarc\OneDrive\Documents\BMEG591E_Genome_Informatics\Group_Project\Mle_sequence.fasta cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Group_Project

pscp -P 22 C:\Users\cbarc\OneDrive\Documents\BMEG591E_Genome_Informatics\Group_Project\mle_RefSeq_genome.fasta cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Group_Project
```

## 1 Introduction

### 1.1 Motivation

From ‘primordial soup’ to the vast array of biodiversity we see today, evolution and these origins of multicellularity still fascinate and elude scientists[1].  Accounts of the number of independent events that led to multicellularity differ amongst the scientific community[2], although there is a consensus that this happened once in the Animalia/Metazoan lineage[2,3]. Advances in sequencing and reduction in cost has led to increase in whole genomic sequences of non-bilateran animal species providing insight into the molecular mechanisms that govern multicellularity[4].

Ctenophores have been proposed as a model organism for understanding the evolutionary mechanisms of multicellularity in animals[4,5], however their phylogenetic placement is still widely debated[5–7]. Ctenophores are a gelatinous phylum of marine metazoans with approximately 150 known species that form a clade of pre-bilateran animals[8].

![Schema of phylogenetic position of Mnemiopsis leidyi](https://github.com/cbarcl01/CB_BMEG591E-repository/blob/master/Group_Project/Mle.jpg)

### 1.2 Original Study


overview of the original study, the scope of your re-analysis, and why you chose it

### 1.3 Activities

## 2 Method

integrated data processing, QC, analysis, results, graphs, and other data, as well as written explanations for what is being done and why, and interpretations of results. This should flow in chronological order (e.g. starting with fastqs and ending with the last graph).

I have downloaded the assembled reference genome as well as the scaffolds. First I will create an Index from the reference genome to use to align the scaffolds to the assembled genome. In theory this should be close to 100% alignment as the reference is built from these scaffolds. 

### 2.1 Preparation and QC

**Download datasets**


**Create an Index**

```{bash, eval=FALSE}
bowtie2-build /home/cbarcl01/Group_Project/Mnemiopsis_leidyi.MneLei_Aug2011.dna.nonchromosomal.fa.gz MleIndex
```

**Investigate file**

First I want to confirm if the number of contigs match the publication:

```{bash, eval=FALSE}
grep -c "^>" AGCP01.fasta
```

*Output: 24884*

**Assess GC Content**

As the raw data was not available in the time constraints, scaffolds in .fasta were used. In this instance Fastqc cannot be run so an alternative script was run to assess the GC content.

```{bash, eval=FALSE}

```


**Alignment**

```{bash, eval=FALSE}
#Unzip fas_nt.ga format
gunzip /home/cbarcl01/Group_Project/AGCP01.1.fsa_nt. > /home/cbarcl01/Group_Project/AGCP01.fasta 

bowtie2 -f -x /home/cbarcl01/Group_Project/MleIndex \ -U /home/cbarcl01/Group_Project/AGCP01.fasta  \ -S /home/cbarcl01/Group_Project/AGCP01.sam
```


```{bash, eval=FALSE}
(Gnme_Assignment_1) [cbarcl01@SBME-GI-EDU-SV4 Group_Project]$ wc -l AGCP01.sam
````

*Output*
24882

**Sam to Bam**
```{bash, eval=FALSE}
samtools view -S -b -h alignment.sam > Alignment.bam
```

### 2.2 Annotation


### 2.3 Phylogeny


### 2.4 RNA-seq expression

## Results

integrated data processing, QC, analysis, results, graphs, and other data, as well as written explanations for what is being done and why, and interpretations of results. This should flow in chronological order (e.g. starting with fastqs and ending with the last graph).


## Conclusion

Summarize your findings and contrast this with what the original study found. Remark on anything surprising or anything you would do differently next time.


## Bibliography

1.	Van Gestel J, Tarnita CE. On the origin of biological construction, with a focus on multicellularity. Proc Natl Acad Sci U S A. 2017;114(42):11018-11026. doi:10.1073/pnas.1704631114
2. 	Niklas KJ, Newman SA. The many roads to and from multicellularity. J Exp Bot. 2020;71(11):3247-3253. doi:10.1093/jxb/erz547
3. 	Niklas KJ, Newman SA. The origins of multicellular organisms. Published online 2013. doi:10.1111/ede.12013
4.  Moreland RT, Nguyen AD, Ryan JF, Baxevanis AD. The Mnemiopsis Genome Project Portal: Integrating new gene expression resources and improving data visualization. Database. 2020;2020(1):1-9. doi:10.1093/database/baaa029
5. 	Ryan JF, Pang K, Schnitzler CE, et al. The genome of the ctenophore Mnemiopsis leidyi and its implications for cell type evolution. Science (80- ). 2013;342(6164). doi:10.1126/science.1242592
6. 	Ryan JF, Pang K, Mullikin JC, Martindale MQ, Baxevanis AD. The homeodomain complement of the ctenophore Mnemiopsis leidyi suggests that Ctenophora and Porifera diverged prior to the ParaHoxozoa. Evodevo. 2010;1(1):1-18. doi:10.1186/2041-9139-1-9
7. 	Ryan JF. Did the ctenophore nervous system evolve independently? Zoology. 2014;117(4):225-226. doi:10.1016/j.zool.2014.06.001
8. 	Moroz LL, Kocot KM, Citarella MR, et al. The ctenophore genome and the evolutionary origins of neural systems. Nature. 2014;510(7503):109-114. doi:10.1038/nature13400

