---
title: "Assignment 4: Mappability continues"
output:
  github_document:
    toc: true
    toc_depth: 4
---

# Assignment Overview

The goal of this assignment is to get you acquainted with how the different ways to analyze a file can change the results of the reads' alignment against the reference genome. We will be using only one file: **SRR12506919_subset.fastq.gz**, that can be found under the following path: **/usr/local/sahre/data/assignment_4/**. It will be divided into two parts: 

Part 1: Analyzing different genome builds 

Part 2: Ambiguity in reads mapping

## Important remarks 

 - Remember to be as clear as possible with your answers.

 - Please make sure to explain your thought process behind your code and answers. 

 - If you have used methods suggested on forums, websites, make sure to cite them.

 - If you have not been able to find the answer to a random bug with reasonable effort, please ask on Piazza! We are here to help, the assignments should be a safe environment for you to try new things and learn. 
 
 - From now on, you will need to specify the output of your Rmarkdown as a github document using: **"output:github_document"** instead of the *output:html_document* default. This will create a folder within your working directory with all the plots created throughout your script, which will be used to create an **.md** extension file. Both the directory with the figures and the .md file will be required for you to be evaluated. Make sure to have them on your private repo. To do this, you need to: 
    
    1. Make sure you have cloned your assignments repository to your local computer

    2. Download the Rmd of the assignment and add it to your repository on your local computer

    3. Edit the assignment to fill in the answers and add your R code

    4. Once you are done, knit your file. Double check that at the top of the Rmd under output, it has github_document

    5. Using your favorite github manager (like Github Desktop), commit your changes to your repository. 

    6. Pull to be up-to-date with any changes to your private repository. 

    7. Push the commit you made on step 6 to your repository. 

    8. Check that on your repository online you can see an *.md* extension file and a folder that includes the plots you created for the assignment, and that the plots are visible when viewing the .md file on GitHub. 


## 0. Getting ready 

As always, before we get started we will install the tools required for the assignment. This time, we only need to add: 

  - LiftOver (https://genome.sph.umich.edu/wiki/LiftOver). These is a package designed to change files from a specific coordinates system (i.e., genome build) to another. 
  
  - bedtools (https://bedtools.readthedocs.io/en/latest/). It's a powerful tool to compare genomic positions. 

```{bash, eval=FALSE}

#?# Add liftOver to your conda environment created on A1 1pt
conda install -c bioconda ucsc-liftover

#?# Add bedtools to your conda environment created on A1 1 pt
conda install -c bioconda bedtools

```

Calling libraries for future use in document:

```{r, eval=TRUE}
library(tidyverse)
```

## 1. Differences between genome builds

Your Professor informs you that the Information and Technology (IT) department was able to recover part of your data from the server. Unfortunately, they were not able to recover your pipelines or scripts. Since you were using a pre-made index of the reference genome, you don't remember which genome build was used to map your sequences (hg19 or hg38). You cannot decide if it would be a big deal to use different genome builds for different alignments, at the end you could just make sure they are in the same genome build when you compare them, right? Thus, you decide to investigate if it would make a lot of difference to use a different version to assess what varies when you align the same reads to different genome-builds.


### a. SE alignment against hg38 and hg19 

```{bash, eval=FALSE}

## Pre-made indexes can be found here: 
## hg19 index: /usr/local/share/indexes/hg19_bowtie2_index
## hg38 index: /usr/local/share/indexes/hg38_bowtie2_index
## Recall that this is the fastq to be used throughput: /usr/local/sahre/data/assignment_4/SRR12506919_subset.fastq.gz

#?# Perform a single-end alignment using bowtie2 against the pre-made index of the hg38 genome build - 2 pt

bowtie2 -x /usr/local/share/indexes/hg38_bowtie2_index \ -U /usr/local/share/data/assignment_4/SRR12506919_subset.fastq.gz \ -S /home/cbarcl01/Assignment_4/hg38_SEBowtie.sam


667557 reads; of these:
  667557 (100.00%) were unpaired; of these:
    52702 (7.89%) aligned 0 times
    486405 (72.86%) aligned exactly 1 time
    128450 (19.24%) aligned >1 times
92.11% overall alignment rate


#?# Perform a single-end alignment using bowtie2 against the pre-made index of the hg19 genome build - 2 pt

bowtie2 -x /usr/local/share/indexes/hg19_bowtie2_index \ -U /usr/local/share/data/assignment_4/SRR12506919_subset.fastq.gz \ -S /home/cbarcl01/Assignment_4/hg19_SEBowtie.sam


667557 reads; of these:
  667557 (100.00%) were unpaired; of these:
    58448 (8.76%) aligned 0 times
    498298 (74.65%) aligned exactly 1 time
    110811 (16.60%) aligned >1 times
91.24% overall alignment rate


```


### b. Making the files comparable

Before you can start comparing the files, you realize you need to translate them to the same genomic coordinate system. If you don't do this and try to find mismatches between the files you will find a ton, but that wouldn't mean that the reads are mapping to different parts of the genome, just that the coordinates are different (e.g. if there is a 1 bp insertion in hg38 relative to hg19, every base after that insertion will have different coordinates). Hence, you go ahead and use your favorite genome build: hg38. To translate hg19 to hg38, we need to do a couple of steps:

 - Sam to Bam: Convert the alignment file (sam) to binary format (bam), this will facilitate the manipulaiton of the files and will decrease the disk space used substantially. 
 
 
 - Bam to bed: Convert the bam alignment file to bed format, enabling the comparison of the genomic posisions where the reads mapped. 
 
 
 - Change genomic coordinates: Use liftOver to change the alignment file made using the hg19 index to the hg38 coordinates of the genome. 
 

```{bash, eval=FALSE}

## Sam to Bam -------------
#?# Convert the SE alignment performed against hg19  (hg19 alignment) to bam, type the command you used below -1 pt

samtools view -S -b ./hg19_SEBowtie.sam > ./hg19_SEBowtie.bam

#?# Convert the SE alignment performed against hg38 (hg38 alignment) to bam, type the command you used below -1 pt

samtools view -S -b ./hg38_SEBowtie.sam > ./hg38_SEBowtie.bam

## Bam to bed -------------
## Tip: Look into the bedtools bamtobed command
#?# Use bedtools to convert the hg19 alignment bam file to bed format, type the command you used below - 1 pt 

bamToBed hg19_SEBowtie.bam > hg19_SEBowtie.bed

#?# Use bedtools to convert the hg38 alignment bam file to bed format, type the command you used below - 1 pt 

bamToBed < hg38_SEBowtie.bam > hg38_SEBowtie.bed


## LiftOver --------------
#?# Use liftOver to change the hg19 alignment bed file to the hg38 coordinate system, type the command/s you used below - 2 pt
## To do this, you will need the "chain file": /usr/local/share/data/assignment_4/hg19ToHg38.over.chain.gz
## Tip: Look at the liftOver documentation! 

liftOver hg19_SEBowtie.bed /usr/local/share/data/assignment_4/hg19ToHg38.over.chain.gz hg19Tohg38_conversion.bed unMapped

```


### c. Analyzing the differences

Now that both alignments are on the same coordinate system, they are comparable and ready to be analyzed. What you really want to see how individual reads mapped against the two genome builds. Did they map to the same place or different places? To answer this, you need to sort your bed files by read name so that you can identify which bed entries in each file correspond to the same original read. 

```{bash, eval=FALSE}

#?# Using bash commands to sort the transformed hg19 alignment file bed alignment file by read name (column 4), type the command you used below - 2 pt
## Tip: Look at the sort command!

sort -k4 hg19Tohg38_conversion.bed > hg19Tohg38_conversion.sorted.bed | head


chr12   4910797         4910948         SRR12506919.100000      42      -
chr20   38723124        38723275        SRR12506919.100001      42      -
chrY    11311528        11311679        SRR12506919.100002      2       +
chr15   34367903        34368054        SRR12506919.100003      42      -
chr11   991338          991489          SRR12506919.10000       42      +
chr11   2939111         2939262         SRR12506919.100004      42      +
chr9    125706890       125707041       SRR12506919.100005      42      +
chr13   49018776        49018927        SRR12506919.100006      34      +
chr5    70641888        70642039        SRR12506919.100007      1       -
chr17   39599897        39600048        SRR12506919.100009      42      +


#?# Using bash commands, sort the hg38 bed alignment file by read name (column 4), type the command you used below - 2 pt
## Tip: Look at the sort command!

sort -k4 hg38_SEBowtie.bed > hg38_SEBowtie.sorted.bed

```


You were really happy to see a visual representation of your data the last time you talked to your advisor about mapping parameters. You decide to give it a try this time with your merged bed file to answer your two main questions:

- How many reads are there per chromosome and does this differ between genome builds? 

- Do the reads mapped to the same genome region?

### d. Reads per chromosome

Before you get started, you discover that a labmate of yours was comparing the number of reads per chromosome under different conditions and they created a function to make this process more robust (function is below). You are really happy that this seems like the perfect function to plot the diferent number of reads per chromosome in the different genome builds, but there is one problem. The bed files need to be merged into one, before you can use the function. Plus, you realize that the function is very poorly documented and your labmate is AWOL due to midterms, so there is no way he can explain you how the function works. Your Professor asks you to go through the function and document as much as possible the function so future people can use it too (also because she wants to make sure you know what you are doing). 


```{bash, eval=FALSE}


## Merging the files: ---------------
#?# Using the join command on bash, merge the two bed files, so they follow the following format: 
## read_id  chr_hg38  start_hg38  end_hg38  strand_hg38 chr_hg19  start_hg19  end_hg19  strand_hg19 

#?# Type the command you used to merge the files below - 2pt 

join -1 4 -2 4 hg38_SEBowtie.sorted.bed hg19Tohg38_conversion.sorted.bed > merged.bed

#?# Use the head command to view the first 3 rows of your merged file, copy the output below: - 2pt 

head merged.bed
 
SRR12506919.100000 chr12 4910797 4910948 42 - chr12 4910797 4910948 42 -
SRR12506919.100001 chr20 38723124 38723275 42 - chr20 38723124 38723275 42 -
SRR12506919.100002 chrY 11311528 11311679 2 + chrY 11311528 11311679 2 +
SRR12506919.100003 chr15 34367903 34368054 42 - chr15 34367903 34368054 42 -
SRR12506919.100004 chr11_KI270831v1_alt 170158 170309 6 + chr11 2939111 2939262 42 +
SRR12506919.100005 chr9 125706890 125707041 42 + chr9 125706890 125707041 42 +
SRR12506919.100006 chr13 49018776 49018927 35 + chr13 49018776 49018927 34 +
SRR12506919.100007 chr5 70821360 70821511 1 - chr5 70641888 70642039 1 -
SRR12506919.100009 chr17 39599897 39600048 42 + chr17 39599897 39600048 42 +
SRR12506919.100010 chr9 36166474 36166625 42 - chr9 36166474 36166625 42 -

## Copy the merged bed file to your local computer for analysis

pscp -P 22 cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Assignment_4/merged.bed C:\Users\cbarc\OneDrive\Desktop
```

Now that you have the files in the right order, you move your files to your local computer to work on your personal RStudio!


```{r, eval=TRUE}

#?# Go through the function line by line using your merged bed file and your chosen parameters, as if it weren't a function (e.g. set "merged_bed" to the data.frame containing your data, and run each line of the function (you will also need to set the parameters)). Explain in a concise way (using comments) how each line is changing the data. Use functions like head and tail to visualize the data as it's changing. - 4 pt

## reads.per.chr:
# This function takes a merged bed file of two conditions A and B and gives a data.frame of 3 columns: Chr, variable (condition), value (how many reads per chromosome are when using that condition)

## Parameters: 
# merged_bed: refers to the bed file you created on the previous section
# cols2compare=c(2,6): default is column 2 versus 6, which if you followed the format specified when you merged the files, they should correspond to the chromosome column of each read for the two conditions (e.g., hg38 and hg19)
# type.a=c("hg38", "redo"): you should specify a string, that states what is condition A. Defaults are "hg38" and "redo"
# type.b=c("hg19", "noDet"): you should specify a string, that states what is condition B. Defaults are "hg19" and "noDet"


reads.per.chr <- function(merged, cols2compare=c(2,7), type.a=c("hg38", "redo"), type.b=c("hg19", "noDet")){
  
  canonical_chromosomes <- paste0("chr", 1:22) # makes vector for chr1 - chr22 
  
  ## head(canonical_chromosomes)
  ##[1] "chr1" "chr2" "chr3" "chr4" "chr5" "chr6"
  
  chr_subset <- merged[,c(cols2compare[1])] # creates a vector of values from merged that match the first column in cols2compare
  table_chrs1 <- table(chr_subset) # count of number of reads for each chromosome
  
  ## head(chr_subset)
  ##[1] "chr12"                "chr20"                "chrY"                 "chr15"                "chr11_KI270831v1_alt" "chr9" 
  
  chr_subset <- merged[,c(cols2compare[2])] # same as above but for the second column in cols2compare
  table_chrs2 <- table(chr_subset)
  
  
  compare.df <- data.frame(column1=table_chrs1[names(table_chrs1) %in% canonical_chromosomes],
                           column2=table_chrs2[names(table_chrs2) %in% canonical_chromosomes]) # creates a dataframe using table_chrs1 and table_chrs2, but only includes chromosomes included in vector canonical_chromosomes
  
  compare.df <- compare.df[,c(1,2,4)] # drops column 3 as it is identical to column 1
  colnames(compare.df) <- c("Chr",paste0(type.a, "_reads"), paste0(type.b, "_reads")) # creates the following vector "Chr"         "hg38_reads"  "redo_reads"  "hg19_reads"  "noDet_reads" 
  
  compare.df <- melt(compare.df)
  
  return(compare.df)
}
```


```{r, eval=TRUE}
#?# Copy the files from the server to your local computer - 1pt

##pscp -P 22 cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Assignment_4/merged.bed C:\Users\cbarc\OneDrive\Desktop

#?# Load your merged bed file into R suing the *read.csv* function and save it into a data.frame
#?# Type the command you used below  - 1pt

mergedbed <- read.table("C:/Users/cbarc/OneDrive/Desktop/git_temp/CB_BMEG591E-repository/Assignment_4/merged.bed", header=FALSE)

## Change the column names of your merged bed data.frame to: 
# "read_id", "chr_hg38", "start_hg38", "end_hg38", "", "", "strand_hg38", "chr_hg19", "start_hg19", "end_hg19", "strand_hg19" 
#?# Type the command you used below:

names(mergedbed) <- c("read_id", "chr_hg38", "start_hg38", "end_hg38", "scorehg38", "strand_hg38", "chr_hg19", "start_hg19", "end_hg19", "scorehg19", "strand_hg19")
## the 42 - or 6 + etc appear as two columns not one for my output so I changed accordingly

## Load the reshape2 library, install it if you don't already have it! 
## Tip: Use the "packages" tab on the left bottom screen 

library(reshape2)

#?# Run the reads.per.chr on your genome builds merged bed (previously loaded), specify all the parameters following the instructions of the function, type the command used below: - 1.5 pt 

mergedf <-reads.per.chr(mergedbed, type.a="hg38", type.b="hg19")

#?# How many reads were mapped to two different chromosomes? What percent of reads is this? Type the code and the answers for each below. 2 pt


NotEqual <- mergedbed %>%
  filter(chr_hg38 != chr_hg19) ##creates a new dataframe where chr_hg38 and chr_hg19 do not match. 

NotEqualCount <- count(NotEqual) ##counts all values in table just created, so the number of reads that do not match chromosones. 

Total <- count(mergedbed)

Percentage <- (NotEqualCount / Total)*100 ## calculates new variable 'Percentage' where reads not equal is divided by the total and multiplied by 100.

#Answer is:
str(Percentage)

## Using the output data.frame you got from running the reads.per.chr function on your merged bed, create a barplot that: 
## Uses the Chr column for the x-axis
## Useds the value (number of reads) column for the y-axis
## Uses the variable (conditions, also known as different genome builds in this case) column to "fill in" the color 
## Each build should have their own bar (next to each other), they shouldn't be stacked!!
#?# Type the command you used below: - 1.5 pt

ggplot(mergedf, aes(fill = variable, x = Chr , y = value )) + 
  geom_bar(stat = "identity", position = "dodge")

```

#### Which chromosome has the biggest difference between reads? Which genome build had more reads for this chromosome? Answer below - 1 pt

**Answer**
Chromosone 21 has the biggest difference, with the hg38 build having the most reads 


### d. Reads position in the genome builds

```{r, eval=TRUE}

## Using the start position of the reads on both genome builds, create a scatterplot using ggplot2 that: 
## Has the start in the hg38 genome build in the x-axis
## Has the start in the hg19 genome build in the y-axis
## Plots each chromosome in its own subplot (panel) (e.g. see facet_wrap())
## Plots only cases where both reads mapped to the same chromosome
#?# Type the command you used below: - 3 pt
canonical_chromosomes <- paste0("chr", 1:22) # makes vector for chr1 - chr22
Scatter_data <- mergedbed %>%
  filter(chr_hg19 %in% canonical_chromosomes) %>%
  filter(chr_hg38 %in% canonical_chromosomes) %>%
  filter(chr_hg38 == chr_hg19)
  

ggplot(Scatter_data, aes(start_hg38, start_hg19)) + geom_point() + facet_wrap(~chr_hg38)

```





## 2. Ambiguity in reads mapping

You are glad that you have answered most of your burning questions about read mapping and identified some of the things that can go wrong. So, you decide to share your knowledge with your friend. They tell you that they ran the SE alignment following your instructions and were about to share their results, only to find that when repeating the alignment for the same file their results changed! They come to you to help them with your wisdom. Your vast experience leads you to believe that something must have happened when the alignment was performed. 

### a. Redoing the hg38 alignment 

```{bash, eval=FALSE}

#?# Re-run the SE alignment that you performed on 1a against the hg38 genome build, use exactly the same parameters, just change the output name  - 0.5 pt

bowtie2 -x /usr/local/share/indexes/hg38_bowtie2_index \ -U /usr/local/share/data/assignment_4/SRR12506919_subset.fastq.gz \ -S /home/cbarcl01/Assignment_4/secondhg38_SEBowtie.sam

667557 reads; of these:
  667557 (100.00%) were unpaired; of these:
    52702 (7.89%) aligned 0 times
    486405 (72.86%) aligned exactly 1 time
    128450 (19.24%) aligned >1 times
92.11% overall alignment rate


## Change both sam output to bam. Remember to remove the sam files right after it's done!
#?# Type the commands you used to convert the file below  - 0.5 pt

samtools view -S -b ./secondhg38_SEBowtie.sam > ./secondhg38_SEBowtie.bam

#?# Change the bam file to bed, using the betdools bedtobam function, type the command you used for the file below - 0.5 pt

bamToBed < secondhg38_SEBowtie.bam > secondhg38_SEBowtie.bed

#?# Sort the file by read name (same as you did on part 1, using column 4), type the command you used below - 1 pt

sort -k4 secondhg38_SEBowtie.bed > secondhg38_SEBowtie.sorted.bed

## Because what you really want to see is if and what changed between these bowtie2 runs compared to your first run on Part 1b, you decide to merge each new run file with the original:

#?# Merge the "redo" bed file and the "original" hg38 alignment bed (from part 1c) using the join command, as in part 1c, this time follow this format: 1 pt
## read_id chr_ori  start_ori  end_ori  strand_ori chr_redo  start_redo  end_redo  strand_redo
## NOTE: Remember to save the output!

join -1 4 -2 4 hg38_SEBowtie.sorted.bed secondhg38_SEBowtie.sorted.bed > redo_merged.bed

## Copy the merged bed file to your local computer for analysis

pscp -P 22 cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Assignment_4/redo_merged.bed C:\Users\cbarc\OneDrive\Desktop

```


### b. Analyzing the ambiguity

Your last analysis on the differences between genome build turn out so well, that you want to do the same. You have prepared the files so they are in the same format as needed to run your labmate's *reads.per.chromosome* function, and are ready to see the graph.


```{r, eval=TRUE}


#?# Load your merged bed file into R using the *read.csv* function and save it into a data.frame
#?# Type the command you used below  - 1pt

redoBed <- read.table("C:/Users/cbarc/OneDrive/Desktop/git_temp/CB_BMEG591E-repository/Assignment_4/redo_merged.bed", header=FALSE)

## Change the column names of your merged bed data.frame to: 
## read_id chr_ori  start_ori  end_ori  strand_ori chr_redo  start_redo  end_redo  strand_redo
#?# Type the command you used below:


names(redoBed) <- c("read_id", "chr_ori", "start_ori", "end_ori", "scoreori", "strand_ori", "chr_redo", "start_redo", "end_redo", "scoreredo", "strand_redo")

#?# Run the reads.per.chr on your genome builds merged bed (previously loaded), specify all the parameters following the instructions of the function, type the command used below: - 1.5 pt 

redof <-reads.per.chr(redoBed, type.a="ori", type.b="redo")


#?# How many reads were mapped to two different chromosomes? What percent of reads is this? Type the code and the answers for each below. 2 pt

RedoNE <- redoBed %>%
  filter(chr_ori != chr_redo) ##creates a new dataframe where chr_hg38 and chr_hg19 do not match. 

RedoNECount <- count(RedoNE) ##counts all values in table just created, so the number of reads that do not match chromosones. 

RedoTotal <- count(redoBed)

RedoPercentage <- (RedoNECount / RedoTotal)*100 ## calculates new variable 'Percentage' where reads not equal is divided by the total and multiplied by 100.

str(RedoPercentage)

## Using the output data.frame you got from running the reads.per.chr function on your merged bed, do a barplot that: 
## Uses the Chr column for the x-axis
## Uses the value (number of reads) column for the y-axis
## Uses the variable (conditions, also known as different runs in this case) column to "fill in" the color 
## Each condition must have their own bar, they shouldn't be stacked!!

#?# Type the command you used below: - 1.5 pt

ggplot(redof, aes(fill = variable, x = Chr , y = value )) + 
  geom_bar(position="dodge", stat = "identity")

#?# Do you see differences among the number of reads per chromosome between the two runs? Answer yes or no - 0.5 pt

## NO

```

You are intrigued by the results of your graph and decide to go deeper into the alignment to get a better idea of where the reads mapped within the genome. 

```{r, eval=TRUE}

## Subtract the start position of the original bed from the start position of the redo for all the reads
#?# Type the command used below: - 0.5 pt

difference <- redoBed$start_ori - redoBed$start_redo

## Use the *table* command to tabulate the results from the previous question. Ex. table(a-b)
#?# Type the command you used below: - 0.5 pt

table(difference)

#?# What do you see? How many have a non zero difference in position start? - 0.5 pt
#?# Describe how would you expect a scatterplot comparing the start ends in both runs would look like - 0.5 pt
## x-axis: original run
## y-axis: re-run 
```
#### What do you see? How many have a non zero difference in position start

None

#### Describe how would you expect a scatterplot comparing the start ends in both runs would look like

If plotted, I would expect to see a scatterplot with all points lying along a best fit where y=x

### c. Non-deterministic seeds

You are confused by your friend's results, you don't seem to have the same problem. You ask her for the command she used to run her alignment and you notice a key difference. She included the following flags: **--non-deterministic --seed 3** . You decide to explore what is this command doing and if it would change your data. 



```{bash, eval=FALSE}

#?# Re-run the SE alignment that you performed on 1a against the hg38 genome build, change the output name and add this parameter:* --non-deterministic --seed 3 * - 1 pt

bowtie2 -x /usr/local/share/indexes/hg38_bowtie2_index \ -U /usr/local/share/data/assignment_4/SRR12506919_subset.fastq.gz \ -S /home/cbarcl01/Assignment_4/ND3_hg38_SEBowtie.sam --non-deterministic --seed 3

667557 reads; of these:
  667557 (100.00%) were unpaired; of these:
    52803 (7.91%) aligned 0 times
    486478 (72.87%) aligned exactly 1 time
    128276 (19.22%) aligned >1 times
92.09% overall alignment rate

## Change both sam outputs to bam. Remember to remove the sam files right after it's done!
#?# Type the commands you used to convert the file below  - 0.5 pt

samtools view -S -b ./ND3_hg38_SEBowtie.sam > ./ND3_hg38_SEBowtie.bam

#?# Change the bam file to bed, using the betdools bedtobam function, type the command you used for the file below  - 0.5 pt

bamToBed < ND3_hg38_SEBowtie.bam > ND3_hg38_SEBowtie.bed

#?# Sort the files by read name (same as you did on part 1, using column 4), type the command you used below - 1 pt

sort -k4 ND3_hg38_SEBowtie.bed > ND3_hg38_SEBowtie.sorted.bed


#?# Merge the "non deterministic" bed file and the "original" hg38 alignment bed (part 1c) using the join command, as in part 1c, this time follow this format: - 1 pt
## read_id  chr_ori  start_ori  end_ori  strand_ori chr_nonDet  start_nonDet  end_nonDet  strand_nonDet 
## NOTE: Remember to save the output!

join -1 4 -2 4 hg38_SEBowtie.sorted.bed ND3_hg38_SEBowtie.sorted.bed > ND3_merged.bed

## Copy the merged bed file to your local computer for analysis

pscp -P 22 cbarcl01@gi-edu-sv4.bme.ubc.ca:/home/cbarcl01/Assignment_4/ND3_merged.bed C:\Users\cbarc\OneDrive\Desktop

```

### d. Analyzing the changes


```{r, eval=TRUE}

#?# Load your merged bed file onto R using the *read.csv* function and save it into a data.frame
#?# Type the command you used below  - 1 pt

ND3Bed <- read.table("C:/Users/cbarc/OneDrive/Desktop/git_temp/CB_BMEG591E-repository/Assignment_4/ND3_merged.bed", header=FALSE)

## Change the column names of your merged bed data.frame to: 
## read_id  chr_ori  start_ori  end_ori  strand_ori chr_nonDet  start_nonDet  end_nonDet  strand_nonDet 
#?# Type the command you used below:

names(ND3Bed) <- c("read_id", "chr_ori", "start_ori", "end_ori", "scoreori", "strand_ori", "chr_nonDet", "start_nonDet", "end_nonDet", "scorenonDet", "strand_nonDet")

#?# How many reads were mapped to two different chromosomes? What percent of reads is this? Type the code and the answers for each below. 2 pt

ND3f <-reads.per.chr(ND3Bed, type.a="ori", type.b="nonDet")

ND3_NE <- ND3Bed %>%
  filter(chr_ori != chr_nonDet) ##creates a new dataframe where chr_hg38 and chr_hg19 do not match. 

ND3_NECount <- count(ND3_NE) ##counts all values in table just created, so the number of reads that do not match chromosones. 

ND3_NETotal <- count(ND3Bed)

ND3Percentage <- (ND3_NECount / ND3_NETotal)*100 ## calculates new variable 'Percentage' where reads not equal is divided by the total and multiplied by 100.


## Using the start position of the reads on both alignment runs do a scatterplot in ggplot that: 
## Has the start in the hg38 genome build in the x-axis
## Has the start in the hg19 genome build in the y-axis
## Plots each chromosome in its own subplot (panel) (e.g. see facet_wrap())
## Plots only cases where both reads mapped to the same chromosome
#?# Type the command you used below: - 2 pt


Scatter_ND3 <- ND3Bed %>%
  filter(chr_ori %in% canonical_chromosomes) %>%
  filter(chr_nonDet %in% canonical_chromosomes) %>%
  filter(chr_ori == chr_nonDet)
  

ggplot(Scatter_ND3, aes(start_ori, start_nonDet)) + geom_point() + facet_wrap(~chr_ori)

#?# Explain why this changes when you add the --non-deterministic --seed 3 flags. What is are these flags doing? Why did you get the result you saw in 2b?- 2 pt
## Tip: Look at the bowtie2 documentation!

#?# How do the number of off-diagonal reads and reads mapping to different chromosomes compare between where we mapped to two different genome versions (and then lifted over), versus the use of non-deterministic alignment? What fraction of reads that you found aligned to different chromsomes when using hg19 vs hg38 result from the differences between these two versions? - 3 pts

diagonal1 <- mergedbed %>%
  filter(chr_hg38 %in% canonical_chromosomes, chr_hg19 %in% canonical_chromosomes, start_hg38 != start_hg19)##creates a new dataframe where chr_hg38 and chr_hg19 do not match, for canonical chromosones shown in scatterplots only. 

diagonal1Count <- count(diagonal1) ##counts all values in table. 

diagonal2 <- ND3Bed %>%
  filter(chr_ori %in% canonical_chromosomes,chr_nonDet %in% canonical_chromosomes, start_ori != start_nonDet ) ##creates a new dataframe where start and chr_hg19 do not match, for canonical chromosones shown in scatterplots only.. 

diagonal2Count <- count(diagonal2) ##counts all values in table just created, so the number of reads that do not match chromosones. 


ND3_diag_Percentage <- (diagonal2Count / ND3_NETotal)*100 ## calculates new variable 'Percentage' where reads not equal is divided by the total and multiplied by 100.

conversion_diag_Percentage <-(diagonal1Count / Total)*100

differenceCount <- diagonal1Count - diagonal2Count

```

#### How many reads were mapped to two different chromosomes? What percent of reads is this?

The percentage of reads that mapped to different chromosones is `r ND3Percentage` %


#### Explain why this changes when you add the --non-deterministic --seed 3 flags. What is/are these flags doing? Why did you get the result you saw in 2b?- 2 pt


As described in the Bowtie 2 manual, when a rea can map equally well to a number of options, Bowtie picks a pseudo-random integer to chose which to align to. This pseudo-random integer is based on "read name, nucleotide string, quality string, and the value specified with --seed". This means that if you run bowtie on two reads with the same name, or, were to re-run your alignment, than the same output is generated.

By choosing the non-deterministic option, Bowtie adds another parameter to the function used to generate the pseudo-random integer - the time. Therefore, as we ran this at two different times, the integer will *potentially* be different and reads will align at another, equally appropriate position.

#?# How do the number of off-diagonal reads and reads mapping to different chromosomes compare between where we mapped to two different genome versions (and then lifted over), versus the use of non-deterministic alignment? What fraction of reads that you found aligned to different chromsomes when using hg19 vs hg38 result from the differences between these two versions?


There are more off-diagonal reads (i.e. where x is not equal to y) in the scatterplot for the hg19 conversion alignment to the the hg38, compared to the scatterplot for hg38 non-deterministic. There are `r differenceCount` more off-diagonal reads in the scatter for hg19 conversion than the non-deterministic scenario. The percentage of off-diagonal reads in hg19 conversion scenario is `r conversion_diag_Percentage`, while the percentage of off-diagonal reads in the non deterministic scenario is `r ND3_diag_Percentage`

There are more reads mapping to different chromosones, when aligning to hg19 and converting to hg38 (using Liftover), compared to the reads mapping to different chromosones when aligning using the non-deterministic hg38 option compared to hg38 (with default options).


Please knit your Rmd file to github_document and include both in your submission.
Successful knitting to github_document - 2 pts

